### PR TITLE
feat: [CI-8700]: gitness CI integration

### PR DESCRIPTION
**Pull request by shaurya-harness**
### Summary

gitness CI integration

#### Screenshots

Screenshot 2023-09-04 at 7 47 14 PM
Screenshot 2023-09-04 at 7 47 39 PM
Screenshot 2023-09-04 at 7 48 42 PM
Screenshot 2023-09-04 at 7 49 10 PM

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/75-ci/pages/get-started-with-ci/InfraProvisioningWizard/InfraProvisioningWizard.tsx [hunk 1]
```diff
@@ -464,8 +485,15 @@ export const InfraProvisioningWizard: React.FC<InfraProvisioningWizardProps> = p
...
-        const fullRepoName = getFullRepoName(selectRepositoryRef.current.repository)
+      if (
+        (selectRepositoryRef.current?.repository || (CODE_ENABLED && selectRepositoryRef.current?.gitnessRepository)) &&
+        configuredGitConnector
+      ) {
+        const fullRepoName = selectRepositoryRef?.current?.repository
+          ? getFullRepoName(selectRepositoryRef.current.repository)
+          : CODE_ENABLED
+          ? selectRepositoryRef.current?.gitnessRepository?.uid || ''
+          : ''
```
Comment (d1wilko) [inline src/modules/75-ci/pages/get-started-with-ci/InfraProvisioningWizard/InfraProvisioningWizard.tsx:496]: this feels a little cleaner like this

```
let fullRepoName = '';

if (selectRepositoryRef?.current?.repository) {
    fullRepoName = getFullRepoName(selectRepositoryRef.current.repository);
} else if (CODE_ENABLED) {
    fullRepoName = selectRepositoryRef.current?.gitnessRepository?.uid || '';
}
```

nested ternarys feel icky to me lol

File: src/modules/75-ci/pages/get-started-with-ci/InfraProvisioningWizard/SelectRepository.tsx [hunk 2]
```diff
@@ -244,22 +303,69 @@ const SelectRepositoryRef = (
             </Container>
           )
         }
-      } else if (repositories && Array.isArray(repositories)) {
-        return repositories.length > 0 ? (
+      } else if (
+        repositories &&
```
Comment (d1wilko) [inline src/modules/75-ci/pages/get-started-with-ci/InfraProvisioningWizard/SelectRepository.tsx:307]: I think you can remove this line - the array test will return false if undefined

File: src/modules/75-ci/pages/get-started-with-ci/InfraProvisioningWizard/SelectRepository.tsx [hunk 3]
```diff
@@ -244,22 +303,69 @@ const SelectRepositoryRef = (
...
+        return repositories && repositories?.length > 0 ? (
           <RepositorySelectionTable repositories={repositories} onRowClick={setRepository} />
         ) : (
           <Text flex={{ justifyContent: 'center' }} padding={{ top: 'medium' }}>
             {getString('noSearchResultsFoundPeriod')}
           </Text>
         )
+      } else if (
+        CODE_ENABLED &&
+        gitnessRepositories &&
```
Comment (d1wilko) [inline src/modules/75-ci/pages/get-started-with-ci/InfraProvisioningWizard/SelectRepository.tsx:320]: i think this can be removed

**Review by d1wilko — CHANGES_REQUESTED** [review]
Some comments but looks good to me for the most part!

I'd ask @vardanbansal-harness to have a look aswell incase I missed things :)

File: src/modules/10-common/components/ReferenceSelect/ReferenceSelect.tsx [hunk 4]
```diff
@@ -326,6 +346,7 @@ export function MultiTypeReferenceInput<T extends MinimalObject>(props: MultiTyp
     return (
       <ReferenceSelect
         {...restProps}
+        useGitProviderComponent={rest.useGitProviderComponent}
```
[the remark this answers is not in the document]
Comment (shaurya-harness) [reply src/modules/10-common/components/ReferenceSelect/ReferenceSelect.tsx:349]: yes, updated this

File: src/modules/75-ci/pages/get-started-with-ci/InfraProvisioningWizard/InfraProvisioningWizard.tsx (hunk 1 above)
Comment (shaurya-harness) [reply src/modules/75-ci/pages/get-started-with-ci/InfraProvisioningWizard/InfraProvisioningWizard.tsx:496]: makes sense

File: src/modules/75-ci/pages/get-started-with-ci/InfraProvisioningWizard/SelectRepository.tsx (hunk 3 above)
Comment (shaurya-harness) [reply src/modules/75-ci/pages/get-started-with-ci/InfraProvisioningWizard/SelectRepository.tsx:320]: agreed

**Review by d1wilko — DISMISSED** [review]
LGTM - but worth getting another set of eyes on it :)

**Review by tan-nhu — DISMISSED** [review]

**Review by d1wilko — APPROVED** [review]

File: src/modules/70-pipeline/components/PipelineStudio/RightBar/RightBarUtils.tsx [hunk 5]
```diff
@@ -123,54 +134,63 @@ export const renderConnectorAndRepoName = ({
       ? connectorWidth + runtimeInputGearWidth
       : connectorWidth
 
+  const renderConnectorField = (): JSX.Element => {
```
Comment (vardanbansal-harness) [inline src/modules/70-pipeline/components/PipelineStudio/RightBar/RightBarUtils.tsx:137]: Is this code duplicated in `CICodebaseInputSetForm` as well?

File: src/modules/75-ci/pages/get-started-with-ci/InfraProvisioningWizard/InfraProvisioningWizard.tsx [hunk 6]
```diff
@@ -221,13 +226,14 @@ export const InfraProvisioningWizard: React.FC<InfraProvisioningWizardProps> = p
...
     selectRepositoryRef.current?.repository,
+    selectRepositoryRef.current?.gitnessRepository,
     yamlVersion
   ])
 
   const constructV0PipelinePayloadWithCodebase = React.useCallback(
-    (repository: UserRepoResponse, pipelineName?: string): string => {
-      const { name: repoName, namespace } = repository
-      if (!repoName || !namespace || !configuredGitConnector?.identifier) {
+    (repository?: UserRepoResponse, gitnessRepoName?: string, pipelineName?: string): string => {
```
Comment (vardanbansal-harness) [inline src/modules/75-ci/pages/get-started-with-ci/InfraProvisioningWizard/InfraProvisioningWizard.tsx:234]: Maybe abstract `repository?: UserRepoResponse, gitnessRepoName?: string, pipelineName?: string` as a different type? I usually wrap it once number of args gets > 2.

File: src/modules/70-pipeline/components/PipelineStudio/RightBar/RightBarUtils.tsx (hunk 5 above)
Comment (shaurya-harness) [reply src/modules/70-pipeline/components/PipelineStudio/RightBar/RightBarUtils.tsx:137]: CICodebaseInputSetForm renders it a little differently, onchange is different and configureOptionsProps are not used there

File: src/modules/75-ci/pages/get-started-with-ci/InfraProvisioningWizard/Constants.ts [hunk 7]
```diff
@@ -20,6 +20,7 @@ export interface InfraProvisioningWizardProps {
   }
   lastConfiguredWizardStepId?: InfraProvisiongWizardStepId
   enableImportYAMLOption?: boolean
+  dummyGitnessHarnessConnector?: ConnectorInfoDTO
```
[the remark this answers is not in the document]
Comment (shaurya-harness) [reply src/modules/75-ci/pages/get-started-with-ci/InfraProvisioningWizard/Constants.ts:23]: self explanatory 😄

File: src/modules/75-ci/pages/get-started-with-ci/InfraProvisioningWizard/InfraProvisioningWizard.tsx (hunk 6 above)
Comment (shaurya-harness) [reply src/modules/75-ci/pages/get-started-with-ci/InfraProvisioningWizard/InfraProvisioningWizard.tsx:234]: sure, can be done

**Review by d1wilko — APPROVED** [review]

**Review by vardanbansal-harness — APPROVED** [review]

**Comment by shaurya-harness** [comment]
retrigger jest

**Comment by shaurya-harness** [comment]
retrigger prettier

**Comment by shaurya-harness** [comment]
retrigger typecheck

**Comment by shaurya-harness** [comment]
retrigger sonar

**Comment by shaurya-harness** [comment]
retrigger standards

**Comment by shaurya-harness** [comment]
retrigger build

**Comment by shaurya-harness** [comment]
retrigger titlecheck

**Comment by shaurya-harness** [comment]
retrigger fetaurenamecheck

**Comment by shaurya-harness** [comment]
retrigger cypress-rest

**Comment by shaurya-harness** [comment]
retrigger cypress-cd

**Comment by shaurya-harness** [comment]
retrigger cypress-pipeline

**Comment by shaurya-harness** [comment]
retrigger cypress-cv

**Comment by shaurya-harness** [comment]
retrigger coverage

**Comment by shaurya-harness** [comment]
fix prettier

**Comment by shaurya-harness** [comment]
retrigger eslint

File: src/modules/27-platform/connectors/components/ConnectorReferenceField/__tests__/__snapshots__/FormMultiTypeGitProviderAndConnectors.test.tsx.snap [hunk 8]
```diff
@@ -0,0 +1,669 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`FormMultiTypeConnectorField tests intial render 1`] = `
+<div>
```
Comment (vikrant-harness) [inline src/modules/27-platform/connectors/components/ConnectorReferenceField/__tests__/__snapshots__/FormMultiTypeGitProviderAndConnectors.test.tsx.snap:4]: Can we avoid such huge snapshot testing and instead assert on the UI

File: src/modules/70-pipeline/components/InputSetForm/FormikInputSetForm.tsx [hunk 9]
```diff
@@ -324,7 +324,14 @@ export function FormikInputSetForm(props: FormikInputSetFormProps): React.ReactE
   const isPipelineRemote = supportingGitSimplification && storeType === StoreType.REMOTE
   React.useEffect(() => {
     const initialValues = getPipelineData()
-    formikRef.current?.setValues({ ...initialValues, ...storeMetadata })
+    const connectorFromPipelineData = initialValues?.pipeline?.properties?.ci?.codebase?.connectorRef
```
Comment (vikrant-harness) [inline src/modules/70-pipeline/components/InputSetForm/FormikInputSetForm.tsx:327]: NIT : can use get here to avoid optional chainings.

Comment (shaurya-harness) [reply src/modules/70-pipeline/components/InputSetForm/FormikInputSetForm.tsx]: done

File: src/modules/27-platform/connectors/components/ConnectorReferenceField/__tests__/__snapshots__/FormMultiTypeGitProviderAndConnectors.test.tsx.snap (hunk 8 above)
Comment (shaurya-harness) [reply src/modules/27-platform/connectors/components/ConnectorReferenceField/__tests__/__snapshots__/FormMultiTypeGitProviderAndConnectors.test.tsx.snap:4]: removed the basic ones

**Review by d1wilko — APPROVED** [review]
LGTM

**Review by vardanbansal-harness — DISMISSED** [review]

**Review by d1wilko — DISMISSED** [review]
LGTM

**Review by vardanbansal-harness — APPROVED** [review]

**Review by vikrant-harness — DISMISSED** [review]

**Review by d1wilko — APPROVED** [review]
LGTM

**Pull request merged**